hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/ClientImpl.java

```diff
@@ -28,10 +30,14 @@
 
     private final String uuid;
     private final InetSocketAddress socketAddress;
+    private final String name;
+    private final Map<String, String> attributes;
 
-    public ClientImpl(String uuid, InetSocketAddress socketAddress) {
```

Comment:
Food for thought: Isn't it easier if name is just an attribute? So you don't need to pass a name and attributes?

Reply:
These two exposed via two different methods on the interface. I think, it is easier to reason about the code when I follow the same structure on the internal usage.